docs/docsite/rst/user_guide/vault.rst

```diff
@@ -196,120 +219,146 @@ Result::
-
 .. _providing_vault_passwords:
 
 Providing Vault Passwords
 `````````````````````````
 
-Since Ansible 2.4, the recommended way to provide a vault password from the cli is
-to use the :option:`--vault-id <ansible-playbook --vault-id>` cli option.
+When all data is encrypted using a single password the :option:`--ask-vault-pass <ansible-playbook --ask-vault-pass>`
+or :option:`--vault-password-file <ansible-playbook --vault-password-file>` cli options should be used.
 
 For example, to use a password store in the text file :file:`/path/to/my/vault-password-file`:
 
 .. code-block:: bash
 
-    ansible-playbook --vault-id /path/to/my/vault-password-file site.yml
+    ansible-playbook --vault-password-file /path/to/my/vault-password-file site.yml
 
 To prompt for a password:
 
 .. code-block:: bash
 
-    ansible-playbook --vault-id @prompt site.yml
+    ansible-playbook --ask-vault-pass site.yml
 
 To get the password from a vault password executable script :file:`my-vault-password.py`:
 
 .. code-block:: bash
 
-    ansible-playbook --vault-id my-vault-password.py
+    ansible-playbook --vault-password-file my-vault-password.py
+
+The config option :ref:`DEFAULT_VAULT_PASSWORD_FILE` can be used to specify a vault password file so that the
+:option:`--vault-password-file <ansible-playbook --vault-password-file>` cli option does not have to be
+specified every time.
+
+
+.. _specifying_vault_ids:
 
-The value for :option:`--vault-id <ansible-playbook --vault-id>` can specify the type of vault id (prompt, a file path, etc)
-and a label for the vault id ('dev', 'prod', 'cloud', etc)
+Labelling Vaults
+^^^^^^^^^^^^^^^^
+
+Since Ansible 2.4 the :option:`--vault-id <ansible-playbook --vault-id>` can be used to indicate which vault id
+('dev', 'prod', 'cloud', etc) a password is for as well as how to source the password (prompt, a file path, etc).
+
+By default the vault-id label is only a hint, any values encrypted with the password will be decrypted.
+The config option :ref:`DEFAULT_VAULT_ID_MATCH` can be set to require the vault id to match the vault id
+used when the value was encrypted.
+This can reduce errors when different values are encrypted with different passwords.
 
 For example, to use a password file :file:`dev-password` for the vault-id 'dev':
 
 .. code-block:: bash
 
     ansible-playbook --vault-id dev@dev-password site.yml
 
-To prompt for the 'dev' vault id:
+To prompt for the password for the 'dev' vault id:
 
 .. code-block:: bash
 
     ansible-playbook --vault-id dev@prompt site.yml
 
-*Prior to Ansible 2.4*
+To get the 'dev' vault id password from an executable script :file:`my-vault-password.py`:
+
+.. code-block:: bash
+
+    ansible-playbook --vault-id dev@my-vault-password.py
 
-To be prompted for a vault password, use the :option:`--ask-vault-pass <ansible-playbook --vault-id>` cli option:
+Since Ansible 2.5 a single executable script can be used to get different password for depending on the vault label.  Scripts with
```

Comment:
This doesn't scan quite right - "to get different password for depending on X" - but I'm not sure exactly what you meant here. Maybe: "Since Ansible 2.5 a single executable script can use vault IDs to get multiple passwords."?

Reply:
You're right, this is confusing because it's really about the implementation of the script but it's mixed in with the "how to tell ansible which script to use" section.

I've re-written this part and broken it out into it's own section after the Multiple Vault Passwords section.

Hopefully it's clearer now.